### DISCUSSION TITLE
Interpreting Security vulnerability Response data.

### DISCUSSION THREAD

**Discussion by naughtron** [community/community#24824] [Apps, API and Webhooks]
<p>Good day, </p>
<p>I have a working PoC script that is able to hit one of my public repositories with the following: </p>
<pre><code>url = 'https://api.github.com/graphql'
query = 'query{repository(owner:"Naughtron", name:"test_sec_alerts") {vulnerabilityAlerts(first:100) {nodes {id}}}}'
headers = {"Authorization": "Bearer &lt;VALID_TOKEN_HERE&gt;", "Accept": "application/vnd.github.vixen-preview+json"}
</code></pre>
<p>I get back the following JSON blob: </p>
<pre><code>{"data":{"repository":{"vulnerabilityAlerts":{"nodes":[{"id":"MDI4OlJlcG9zaXRvcnlWdWxuZXJhYmlsaXR5QWxlcnQxMzY4NzYwMDg="},{"id":"MDI4OlJlcG9zaXRv
cnlWdWxuZXJhYmlsaXR5QWxlcnQxMzc5OTQ0OTQ="},{"id":"MDI4OlJlcG9zaXRvcnlWdWxuZXJhYmlsaXR5QWxlcnQxMzc5OTQ0OTU="},{"id":"MDI4OlJlcG9zaXRvcnlWdWxuZXJh
YmlsaXR5QWxlcnQxMzc5OTQ0OTc="},{"id":"MDI4OlJlcG9zaXRvcnlWdWxuZXJhYmlsaXR5QWxlcnQxMzc5OTQ0OTg="}]}}}}
</code></pre>
<p>If I base64 decode one of the results, for example: </p>
<pre><code>MDI4OlJlcG9zaXRvcnlWdWxuZXJhYmlsaXR5QWxlcnQxMzY4NzYwMDg=
</code></pre>
<p>I am presented with: </p>
<pre><code>028:RepositoryVulnerabilityAlert136876008
</code></pre>
<p>From this point what can I do with that information? Is there a way to ultimately get information as to the actual library and its location in the repository as we get via the UI? </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/7/7785947b3cdfb25c7abe22418cda546ff2aec6a3.png" data-download-href="/uploads/short-url/h3l0NiBNbEao5gCjOVBbMH6eR35.png?dl=1" title="Screen Shot 2019-10-29 at 11.48.15 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/776783/181087006-da54f8a8-6b32-4b98-8027-a47ea21ce25e.png" alt="Screen Shot 2019-10-29 at 11.48.15 AM.png" data-base62-sha1="h3l0NiBNbEao5gCjOVBbMH6eR35" width="690" height="317" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/7/7785947b3cdfb25c7abe22418cda546ff2aec6a3_2_690x317.png, https://aws1.discourse-cdn.com/github/optimized/2X/7/7785947b3cdfb25c7abe22418cda546ff2aec6a3_2_1035x475.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/7/7785947b3cdfb25c7abe22418cda546ff2aec6a3_2_1380x634.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/7/7785947b3cdfb25c7abe22418cda546ff2aec6a3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-29 at 11.48.15 AM.png</span><span class="informations">2090×962 239 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by lee-dohm** [comment, marked as answer]
<p>Node IDs of all types are intended to be opaque. There is nothing you can get from them without requesting more information from the API. You can find <a href="https://developer.github.com/v4/object/repositoryvulnerabilityalert/" rel="nofollow noopener">the other fields that are available from the <code>RepositoryVulnerabilityAlert</code> object</a> (besides <code>id</code>) in the GraphQL API documentation.</p>
<p>For example, you could use this query:</p>
<pre><code>{
  repository(owner: "Naughtron", name: "test_sec_alerts") {
    vulnerabilityAlerts(first: 100) {
      nodes {
        securityVulnerability {
          package {
            ecosystem
            name
          }
          vulnerableVersionRange
          firstPatchedVersion {
            identifier
          }
        }
      }
    }
  }
}
</code></pre>
<p>I hope that helps!</p>
